presto-main/src/test/java/io/prestosql/sql/rewrite/ShowQueriesRewriteTest.java

```diff
@@ -0,0 +1,240 @@
+import io.prestosql.execution.warnings.WarningCollector;
+import io.prestosql.metadata.Catalog;
+import io.prestosql.metadata.CatalogManager;
+import io.prestosql.metadata.InMemoryNodeManager;
+import io.prestosql.metadata.InternalNodeManager;
+import io.prestosql.metadata.Metadata;
+import io.prestosql.security.AccessControlManager;
+import io.prestosql.spi.connector.CatalogSchemaTableName;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.Connector;
+import io.prestosql.spi.connector.ConnectorMetadata;
+import io.prestosql.spi.connector.ConnectorTableMetadata;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.connector.SchemaTableName;
+import io.prestosql.spi.security.AccessDeniedException;
+import io.prestosql.spi.security.SystemAccessControl;
+import io.prestosql.spi.security.SystemAccessControlFactory;
+import io.prestosql.spi.security.SystemSecurityContext;
+import io.prestosql.spi.transaction.IsolationLevel;
+import io.prestosql.sql.analyzer.FeaturesConfig;
+import io.prestosql.sql.parser.SqlParser;
+import io.prestosql.testing.TestingMetadata;
+import io.prestosql.transaction.TransactionManager;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.security.Principal;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import static io.prestosql.connector.CatalogName.createInformationSchemaCatalogName;
+import static io.prestosql.connector.CatalogName.createSystemTablesCatalogName;
+import static io.prestosql.metadata.MetadataManager.createTestMetadataManager;
+import static io.prestosql.operator.scalar.ApplyFunction.APPLY_FUNCTION;
+import static io.prestosql.spi.security.AccessDeniedException.denyShowColumnsMetadata;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.sql.ParsingUtil.createParsingOptions;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static io.prestosql.transaction.InMemoryTransactionManager.createTestTransactionManager;
+import static io.prestosql.transaction.TransactionBuilder.transaction;
+import static java.util.Objects.requireNonNull;
+
+@Test(singleThreaded = true)
+public class ShowQueriesRewriteTest
+{
+    private static final String CATALOG = "c";
+    private static final CatalogName CATALOG_NAME = new CatalogName(CATALOG);
+    private static final Session CATALOG_SESSION = testSessionBuilder()
+            .setCatalog(CATALOG)
+            .setSchema("s1")
+            .build();
+
+    private static final SqlParser SQL_PARSER = new SqlParser();
+
+    private TransactionManager transactionManager;
+    private AccessControlManager accessControl;
+    private Metadata metadata;
+
+    private static SchemaTableName allowedTabled = new SchemaTableName("s1", "t1");
```

Comment:
`t1` -> `accessible_table` and `t2` -> `non_accessible_table`

Reply:
sure